nixos/modules/services/networking/aria2.nix

```diff
@@ -120,17 +154,20 @@ in
           touch "${sessionFile}"
         fi
         cp -f "${settingsFile}" "${settingsDir}/aria2.conf"
+        chmod +w "${settingsDir}/aria2.conf"
```

Comment:
Why?

Reply:
To be honest, I don't know why it worked before. Files in the nix store are not writable and copying the file from the nix store to some other directory doesn't change that. See the `set -x` test output here:
```
machine # [    8.628432] aria2-pre-start[794]: + cp -f /nix/store/9pysl0bgc35wagb4qyrni43a0sgmjfmm-aria2.conf /var/lib/aria2/aria2.conf
machine # [    8.631637] systemd[1]: Finished Permit User Sessions.
machine # [    8.644510] systemd[1]: Started Getty on tty1.
machine # [    8.648132] systemd[1]: Reached target Login Prompts.
machine # [    8.650737] aria2-pre-start[794]: + ls -la /var/lib/aria2
machine # [    8.697284] aria2-pre-start[801]: total 16
machine # [    8.698995] aria2-pre-start[801]: drwxrwx--- 3 aria2 aria2 4096 Apr 16 18:57 .
machine # [    8.702515] aria2-pre-start[801]: drwxr-xr-x 8 root  root  4096 Apr 16 18:57 ..
machine # [    8.704837] aria2-pre-start[801]: -r--r--r-- 1 aria2 aria2  307 Apr 16 18:57 aria2.conf
machine # [    8.706943] aria2-pre-start[801]: -rw-r--r-- 1 aria2 aria2    0 Apr 16 18:57 aria2.session
machine # [    8.709370] aria2-pre-start[801]: drwxrwx--- 2 aria2 aria2 4096 Apr 16 18:57 Downloads
machine # [    8.712158] aria2-pre-start[794]: + chmod +w /var/lib/aria2/aria2.conf
machine # [    8.715914] aria2-pre-start[794]: + ls -la /var/lib/aria2
machine # [    8.731449] aria2-pre-start[803]: total 16
machine # [    8.732721] aria2-pre-start[803]: drwxrwx--- 3 aria2 aria2 4096 Apr 16 18:57 .
machine # [    8.734577] aria2-pre-start[803]: drwxr-xr-x 8 root  root  4096 Apr 16 18:57 ..
machine # [    8.736273] aria2-pre-start[803]: -rw-r--r-- 1 aria2 aria2  307 Apr 16 18:57 aria2.conf
machine # [    8.738103] aria2-pre-start[803]: -rw-r--r-- 1 aria2 aria2    0 Apr 16 18:57 aria2.session
machine # [    8.739719] aria2-pre-start[803]: drwxrwx--- 2 aria2 aria2 4096 Apr 16 18:57 Downloads
```
So I need to chmod in order to add the rpc secret.